src/inventory_ui.cpp

```diff
@@ -829,35 +829,53 @@ int game::inv_for_salvage(const std::string &title, const salvage_actor& actor )
     return display_slice(reduced_inv, title);
 }
 
-item *game::inv_map_for_liquid(const item &liquid, const std::string &title)
+std::pair< int, item* > game::inv_map_splice( item_filter filter, const std::string &title )
+{
+    return inv_map_splice( filter, filter, title );
+}
+
+std::pair< int, item* > game::inv_map_splice( item_filter inv_filter, item_filter ground_filter, const std::string &title )
 {
-    auto here = m.i_at(g->u.posx(), g->u.posy());
+    constexpr char first_invlet = '0';
+    constexpr char last_invlet = '9';
+
+    auto here = m.i_at( g->u.pos3() );
     typedef std::vector< std::list<item> > pseudo_inventory;
     pseudo_inventory grounditems;
     indexed_invslice grounditems_slice;
-    std::vector<item *> ground_containers;
+    std::vector<item *> ground_selectables;
 
     std::set<std::string> dups;
     for( auto candidate = here.begin(); candidate != here.end(); ++candidate ) {
-        if( candidate->get_remaining_capacity_for_liquid( liquid ) > 0 ) {
-            if( dups.count( candidate->tname() ) == 0 ) {
```

Comment:
By removing this check on `dups`, the set is not used anymore. There is still an insertion below, but it's never read, so why bother with. `dups` should be removed completely.